cockatrice/src/carddatabase.cpp

```diff
@@ -609,57 +597,59 @@ bool CardDatabase::saveToFile(const QString &fileName, bool tokens)
     return true;
 }
 
-void CardDatabase::emitCardListChanged()
-{
-    emit cardListChanged();
-}
-
-LoadStatus CardDatabase::loadCardDatabase(const QString &path, bool tokens)
+LoadStatus CardDatabase::loadCardDatabase(const QString &path)
 {
     LoadStatus tempLoadStatus = NotLoaded;
     if (!path.isEmpty())
-        tempLoadStatus = loadFromFile(path, tokens);
-
-    if (tempLoadStatus == Ok) {
-        SetList allSets;
-        QHashIterator<QString, CardSet *> setsIterator(sets);
-        while (setsIterator.hasNext())
-            allSets.append(setsIterator.next().value());
-        allSets.sortByKey();
-
-        if(!tokens)
-            checkUnknownSets();
-        emit cardListChanged();
-    }
-
-    if (!tokens)
-        loadStatus = tempLoadStatus;
+        tempLoadStatus = loadFromFile(path);
 
-    qDebug() << "loadCardDatabase(): Path =" << path << "Tokens =" << tokens << "Status =" << loadStatus;
+    qDebug() << "[CardDatabase] loadCardDatabase(): Path =" << path << "Status =" << tempLoadStatus << "Cards =" << cards.size() << "Sets=" << sets.size();
 
     return tempLoadStatus;
 }
 
-LoadStatus CardDatabase::loadCardDatabase()
+LoadStatus CardDatabase::loadCardDatabases()
 {
-    return loadCardDatabase(settingsCache->getCardDatabasePath(), false);
-}
+    qDebug() << "CardDatabase::loadCardDatabases start";
+    // clean old db
+    clear();
+    // load main card database
+    loadStatus = loadCardDatabase(settingsCache->getCardDatabasePath());
+    // laod tokens database
+    loadCardDatabase(settingsCache->getTokenDatabasePath());
+    // load custom card databases
+    QDir dir(settingsCache->getCustomCardDatabasePath());
+    foreach(QString fileName, dir.entryList(QStringList("*.xml"), QDir::Files | QDir::Readable, QDir::Name | QDir::IgnoreCase))
+    {
+        loadCardDatabase(dir.absoluteFilePath(fileName));
```

Comment:
We should add error handling and logging to this to help people debug bad custom sets